funnel/views/commentvote.py

```diff
@@ -1,52 +1,58 @@
 # -*- coding: utf-8 -*-
 
-from flask import redirect, g, flash
+from flask import redirect, g, flash, abort
 from coaster.views import jsonp, load_models
-from baseframe import _
+from baseframe import _, forms
 
 from .. import app, lastuser
 from ..models import db, Profile, ProposalSpace, ProposalSpaceRedirect, Proposal, ProposalRedirect, Comment
 
 
-# FIXME: This voting method uses GET but makes db changes. Not correct. Should be POST
-@app.route('/<space>/<proposal>/voteup', subdomain='<profile>')
+@app.route('/<space>/<proposal>/voteup', subdomain='<profile>', methods=['POST'])
 @lastuser.requires_login
 @load_models(
     (Profile, {'name': 'profile'}, 'g.profile'),
     ((ProposalSpace, ProposalSpaceRedirect), {'name': 'space', 'profile': 'profile'}, 'space'),
     ((Proposal, ProposalRedirect), {'url_name': 'proposal', 'proposal_space': 'space'}, 'proposal'),
     permission='vote-proposal', addlperms=lastuser.permissions)
 def proposal_voteup(profile, space, proposal):
+    csrf_form = forms.Form()
+    if not csrf_form.validate_on_submit():
+        abort(401)
     proposal.votes.vote(g.user, votedown=False)
     db.session.commit()
     flash(_("Your vote has been recorded"), 'info')
     return redirect(proposal.url_for())
```

Comment:
Redirects should be with status 303 given this is a POST. Also, return JSON status for XHR requests.